Sources/RoutingKit/TrieRouter/TrieRouterNode.swift

```diff
@@ -0,0 +1,106 @@
+/// A single node of the ``TrieRouter``s trie tree of routes.
+final class TrieRouterNode<Output: Sendable>: Sendable, CustomStringConvertible {
+    /// Describes a node that has matched a parameter or anything
+    struct Wildcard: Sendable {
+        let parameter: String?
+        let explicitlyIncludesAnything: Bool
+
+        let node: TrieRouterNode
+
+        init(node: TrieRouterNode, parameter: String? = nil, explicitlyIncludesAnything: Bool = false) {
+            self.node = node
+            self.parameter = parameter
+            self.explicitlyIncludesAnything = explicitlyIncludesAnything
+        }
+
+        static func anything(_ node: TrieRouterNode) -> Wildcard {
+            Wildcard(node: node, explicitlyIncludesAnything: true)
+        }
+
+        static func parameter(_ node: TrieRouterNode, named name: String) -> Wildcard {
+            Wildcard(node: node, parameter: name)
+        }
+
+        func copyWith(
+            node: TrieRouterNode? = nil,
+            parameter: String? = nil,
+            explicitlyIncludesAnything: Bool? = nil
+        ) -> Wildcard {
+            Wildcard(
+                node: node ?? self.node,
+                parameter: parameter ?? self.parameter,
+                explicitlyIncludesAnything: explicitlyIncludesAnything ?? self.explicitlyIncludesAnything
+            )
+        }
+    }
+
+    /// All constant child nodes.
+    let constants: [String: TrieRouterNode]
+
+    /// Wildcard child node that may be a named parameter or an anything
+    let wildcard: Wildcard?
+
+    /// Catchall node, if one exists.
+    /// This node should not have any child nodes.
+    let catchall: TrieRouterNode?
+
+    /// This node's output
+    let output: Output?
+
+    /// Creates a new `RouterNode`.
```

Comment:
Indefinitely perpetuating Tanner's constitutional incapacity to update comments that date back to Vapor 2 is not preferred 😛.
```suggestion
    /// Creates a new ``TrieRouterNode``.
```